include/entities/entity_factory.hpp

```diff
@@ -29,19 +32,22 @@ class EntityFactory {
         return renderableFactory;
     };
 
-    int createPlayer(int player_id, Character character, int x = -2000, int y = -2000) const;
-    int createGorilla(int player_id) const;
-    int createPanda(int player_id) const;
-    int createCheetah(int player_id) const;
-    int createElephant(int player_id) const;
-    int createCritter(double x_pos, double y_pos) const;
-    int createSpawner(double x_pos, double y_pos, std::vector<GadgetType> available_spawns, int respawn_timer,
+    EntityComponents createPlayer(int player_id, Character character, int x = -2000, int y = -2000) const;
+    EntityComponents createCritter(double x_pos, double y_pos) const;
+    EntityComponents createImage(std::string path, int x_pos, int y_pos, int x_scale, int y_scale, double relative_modifier, Layers layer, int alpha);
+    EntityComponents createPlatform(double x_pos, double y_pos, double x_scale, double y_scale, double relative_modifier, std::string path, int alpha);
+    EntityComponents createTrophy(int x, int y, int x_scale, int y_scale, double relative_modifier, Layers layer, int alpha);
+    EntityComponents createReadySign(int x, int y, int x_scale, int y_scale, double relative_modifier, Layers layer, int alpha);
+    std::vector<EntityComponents> createButton(std::string text, Color text_color, int font_size,
+        std::string texture_path, int x, int y, int x_scale, int y_scale, 
+        int alpha, double relative_modifier, std::function<void ()> on_click);
+    EntityComponents createText(std::string text, Color color, int font_size, int x, int y, int x_scale, int y_scale, double relative_modifier);
+
+    int addToEntityManager(EntityComponents entity_components, std::string scene_tag ,std::optional<std::pair<int,bool>> parent_opt = std::nullopt);
```

Comment:
This is a quite handy function, but it should be used lightly, only when you can't access the EntityManager or got some EntityComponents from the EntityFactor. You should comment that above the function.

The function also doesn't correctly reflect the entity_manager, the scene_tag should be optional. When you are already in a scene such as in the Scene::start() or a system, you should let the entity_manager set the scene_tag for you.